android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt

```diff
@@ -568,10 +587,135 @@ class AirshipPlugin : MethodCallHandler, FlutterPlugin {
 
     private fun openPreferenceCenter(call: MethodCall, result: Result) {
         val preferenceCenterID = call.arguments as String
-        PreferenceCenter.shared().open(preferenceCenterId = preferenceCenterID)
+
+        val sharedPref: SharedPreferences = context.getSharedPreferences("shared_preferences", Context.MODE_PRIVATE)
+        val enabled = sharedPref.getBoolean(AUTO_LAUNCH_PREFERENCE_CENTER_KEY, true)
+
+        if (enabled) {
+            PreferenceCenter.shared().open(preferenceCenterID)
+        } else {
+            EventManager.shared.notifyEvent(ShowPreferenceCenterEvent(preferenceCenterID))
+        }
         result.success(null)
     }
 
+    private fun getSubscriptionLists(call: MethodCall, result: Result) {
+
+        val subscriptionTypes = call.arguments as List<String>
+
+        val subscriptionLists = mutableMapOf<String, Any>()
+
+        scope.launch {
+            val r1 = UAirship.shared().contact.getSubscriptionLists(true)
+            suspendCoroutine<Unit> { continuation ->
```

Comment:
I'm pretty sure this will end up making the requests and waiting for results in sequence, since the `suspendCoroutine`s will suspend this coroutine until the result callback is called. I think `async` seems like a better fit here, so that we can fetch in parallel. It should also be fine to just block on the `PendingResult`s instead of using result callbacks if we also move off of main dispatcher.

```kotlin
scope.launch(Dispatchers.Main) {
    val subscriptionLists = withContext(Dispatchers.IO) {
        val channelSubs = async {
            UAirship.shared().channel.getSubscriptionLists(true).get()
                ?.toList()
                ?: emptyList()
        }
        val contactSubs = async {
            UAirship.shared().contact.getSubscriptionLists(true).get()
                ?.mapValues { it.value.map(Scope::toString) }
                ?: emptyMap()
        }
       
       // Block until both requests complete and return a map
        mapOf(
            "channel" to channelSubs.await(),
            "contact" to contactSubs.await()
        )
    }

    // Callback on main dispatcher with result
    result.success(subscriptionLists)
}
```